### PR TITLE
En 675 page configuration endpoints

### PR DESCRIPTION
**Pull request by stefanopuddu**
- add page configuration endpoints
`/pages/<pageCode>/configuration  [GET]`
`/pages/<pageCode>/widgets/ [GET]`
`/pages/<pageCode>/widget/<frameId> [GET] [PUT] [DELETE]`
- add `WidgetValidatorFactory` and related cms `content_viewer` implementation

### CODE REVIEW

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
ok to test

**Comment by entando-service** [comment]
All is well

File: engine/src/main/java/org/entando/entando/aps/system/services/page/model/PageConfigurationDto.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+public class PageConfigurationDto {
+
+    private String code; //
+    private String parentCode; //
+    private String status; //
+
+    private int position; //
+    private String ownerGroup;//
+
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/aps/system/services/page/model/PageConfigurationDto.java]: Remove trailing //

File: engine/src/main/java/org/entando/entando/aps/system/services/page/model/PageConfigurationDto.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        }
+        List<WidgetConfigurationDto> widgetDtoList = new ArrayList<>();
+        for (Widget widget : widgets) {
+            if (null == widget) {
+                widgetDtoList.add(null);
+            } else {
+                widgetDtoList.add(new WidgetConfigurationDto(widget.getType().getCode(), widget.getConfig()));
+            }
+        }
+        this.setWidgets(widgetDtoList.toArray(new WidgetConfigurationDto[widgetDtoList.size()]));
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/aps/system/services/page/model/PageConfigurationDto.java]: Is this recursing? Looks like an infinite loop possibly. `this.widgets = widgetDtoList.toArray(new WidgetConfigurationDto[widgetDtoList.size()])` ?

File: engine/src/main/java/org/entando/entando/web/page/PageConfigurationController.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    }
+
+    public void setPageService(IPageService pageService) {
+        this.pageService = pageService;
+    }
+
+    @RestAccessControl(permission = Permission.SUPERUSER)
+    @RequestMapping(value = "/pages/{pageCode}/configuration", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<?> getPageConfiguration(@PathVariable String pageCode, @RequestParam(value = "status", required = false, defaultValue = IPageService.STATUS_DRAFT) String status) {
+        PageConfigurationDto pageConfiguration = this.getPageService().getPageConfiguration(pageCode, status);
```
Comment (joewhite101) [inline engine/src/main/java/org/entando/entando/web/page/PageConfigurationController.java]: For all of these controller methods lets put in some debug or trace logging to include the payload

File: engine/src/test/java/org/entando/entando/aps/system/services/page/PageService_WidgetIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.entando.entando.aps.system.services.page.model.PageConfigurationDto;
+import org.entando.entando.aps.system.services.page.model.WidgetConfigurationDto;
+import org.entando.entando.web.page.model.WidgetConfigurationRequest;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.junit.Assert.assertThat;
+
+public class PageService_WidgetIntegrationTest extends BaseTestCase {
```
Comment (joewhite101) [inline engine/src/test/java/org/entando/entando/aps/system/services/page/PageService_WidgetIntegrationTest.java]: Remove underscore. No underscores in class names

File: engine/src/test/java/org/entando/entando/aps/system/services/page/PageService_WidgetIntegrationTest.java [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+            assertThat(this.pageService.getWidgetConfiguration(pageToAdd.getCode(), 0, IPageService.STATUS_DRAFT).getCode(), is("login_form"));
+
+        } finally {
+            pageManager.deletePage(pageCode);
+        }
+
+    }
+
+    public void test_remove_page_widget() throws JsonProcessingException, ApsSystemException {
```
Comment (joewhite101) [inline engine/src/test/java/org/entando/entando/aps/system/services/page/PageService_WidgetIntegrationTest.java]: No underscores in method names

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by eugeniosant** [comment]
retest this please

**Comment by entando-service** [comment]
All is well

**Pull request merged**